### DISCUSSION TITLE
Is the Java Runtime really necessary?

### DISCUSSION THREAD

**Discussion by ricardo-valero** [ionic-team/capacitor#4982] [Q&A] [answered]
I **can** open the project and run it with Android Studio since I have de SDK, but is it possible for the command

```bash
cap run android 
```

to run without Java Runtime? or is it on the roadmap? 

I get this error:

```
✖ Running Gradle build - failed!
[error] The operation couldn’t be completed. Unable to locate a Java Runtime.
        Please visit http://www.java.com for information on installing Java.
```

**Comment by jcesarmobile** [comment, marked as answer]
Android development requires a Java runtime, android studio bundles its own. What I do is to set my JAVA_HOME to point to the android studio Java installation so it uses the same.

**Reply by ricardo-valero** [reply]
Oh it makes sense. I was dumb hehe. Thanks 😄 

Maybe they should add to [Capacitor docs](https://capacitorjs.com/docs/getting-started/environment-setup#android-development) something like [NativeScript docs](https://docs.nativescript.org/environment-setup.html#macos-android):

> Add the following lines to your shell profile, usually `~/.bash_profile` or `~/.bashrc`, or if you are using zsh then `~/.zprofile` or `~/.zshrc` config file:
> ```
> export JAVA_HOME=/Applications/Android\ Studio.app/Contents/jre/Contents/Home/
> ```
